src/sfloader/fluid_sffile.h

```diff
@@ -0,0 +1,271 @@
+typedef struct _SFMod SFMod;
+typedef struct _SFGen SFGen;
+typedef struct _SFPresetZone SFPresetZone;
+typedef struct _SFInstZone SFInstZone;
+typedef struct _SFSample SFSample;
+typedef struct _SFInst SFInst;
+typedef struct _SFPreset SFPreset;
+typedef struct _SFData SFData;
+typedef struct _SFChunk SFChunk;
+typedef struct _SFPhdr SFPhdr;
+typedef struct _SFBag SFBag;
+typedef struct _SFIhdr SFIhdr;
+typedef struct _SFShdr SFShdr;
+
+
+struct _SFVersion
+{ /* version structure */
+    unsigned short major;
+    unsigned short minor;
+};
+
+struct _SFMod
+{ /* Modulator structure */
+    unsigned short src; /* source modulator */
+    unsigned short dest; /* destination generator */
+    signed short amount; /* signed, degree of modulation */
+    unsigned short amtsrc; /* second source controls amnt of first */
+    unsigned short trans; /* transform applied to source */
+};
+
+union _SFGenAmount { /* Generator amount structure */
+    signed short sword; /* signed 16 bit value */
+    unsigned short uword; /* unsigned 16 bit value */
+    struct
+    {
+        unsigned char lo; /* low value for ranges */
+        unsigned char hi; /* high value for ranges */
+    } range;
+};
+
+struct _SFGen
+{ /* Generator structure */
+    unsigned short id; /* generator ID */
+    SFGenAmount amount; /* generator value */
+};
+
+struct _SFPresetZone
+{ /* Preset zone structure */
+    unsigned short gen_idx;
+    int gen_count;
+
+    unsigned short mod_idx;
+    int mod_count;
+
+    SFInst *inst;
+
+    fluid_list_t *gen; /* list of generators */
+    fluid_list_t *mod; /* list of modulators */
+};
+
+struct _SFInstZone
+{ /* Instrument zone structure */
+    unsigned short gen_idx;
+    int gen_count;
+
+    unsigned short mod_idx;
+    int mod_count;
+
+    SFSample *sample;
+
+    fluid_list_t *gen; /* list of generators */
+    fluid_list_t *mod; /* list of modulators */
+};
+
+struct _SFSample
+{ /* Sample structure */
+    unsigned short idx;
+
+    char name[21]; /* Name of sample */
+    unsigned char samfile; /* Loaded sfont/sample buffer = 0/1 */
+    unsigned int start; /* Offset in sample area to start of sample */
+    unsigned int end; /* Offset from start to end of sample,
+             this is the last point of the
+             sample, the SF spec has this as the
+             1st point after, corrected on
+             load/save */
+    unsigned int loopstart; /* Offset from start to start of loop */
+    unsigned int loopend; /* Offset from start to end of loop,
+                 marks the first point after loop,
+                 whose sample value is ideally
+                 equivalent to loopstart */
+    unsigned int samplerate; /* Sample rate recorded at */
+    unsigned char origpitch; /* root midi key number */
+    signed char pitchadj; /* pitch correction in cents */
+    unsigned short sampletype; /* 1 mono,2 right,4 left,linked 8,0x8000=ROM */
+};
+
+struct _SFInst
+{ /* Instrument structure */
+    unsigned short idx;
```

Comment:
Perhaps better use `unsigned int` for this index?

Reply:
Yes... no idea why I chose short here...